e2e/imgbuild/imgbuild.go

```diff
@@ -1589,44 +1589,76 @@ func (c *imgBuildTests) testContainerBuildUnderFakerootModes(t *testing.T) {
 	)
 }
 
+func (c *imgBuildTests) testSIFHeaderAndExecute(t *testing.T) {
+	tmpDir, cleanup := e2e.MakeTempDir(t, c.env.TestDir, "test-SIF-header-fields-", "")
+	defer cleanup(t)
+
+	path := fmt.Sprintf("%s/img.sif", tmpDir)
+
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs(path, "docker://busybox"),
+		e2e.ExpectExit(0),
+	)
+
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("sif"),
+		e2e.WithArgs("header", path),
+		e2e.ExpectExit(0,
+			e2e.ExpectOutput(e2e.ContainMatch, "#!/usr/bin/env run-singularity"),
+			e2e.ExpectOutput(e2e.RegexMatch, ".*[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}.*"),
+		))
+
+	cmd := exec.Command(path)
```

Comment:
You should be able to pass parameters to the command as well.  As it is, I think it happens to work because /dev/null is stdin.  If you ran the sif file by itself interactively it would give a prompt.  Is suggest having it echo something and looking for that output in stdout.

Reply:
thanks for the advice! added output string and squashed all commits to one.